test/shared/src/main/scala/zio/test/mock/MockException.scala

```diff
@@ -0,0 +1,15 @@
+package zio.test.mock
+
+import zio.test.{ AssertResult, Assertion }
+
+sealed trait MockException extends Throwable
+
+final case class PredicateFailedException(predicate: AssertResult) extends MockException
+
+final case class UnexpectedCallException(method: Method[Nothing, Any], args: Any) extends MockException
+
+final case class WrongCallException(
+  expected: Method[Nothing, Any],
```

Comment:
You should make `AnyMethod`, a `sealed trait` that is extended only bby `Method[A, B]`, so you have a saner way to refer to these "any method" handles.

Reply:
Is there any reason to drop type information here? I made the exceptions parametrized in `A` and `B` and could drop type casts to `Method[Nothing, Any]` and everything still works.